lib/motions/general-motions.coffee

```diff
@@ -150,6 +150,7 @@ class MoveRight extends Motion
   moveCursor: (cursor, count=1) ->
     _.times count, =>
       cursor.moveRight() unless cursor.isAtEndOfLine()
+      cursor.moveRight() if atom.config.get('vim-mode.wrapLeftRightMotion') and cursor.isAtEndOfLine()
```

Comment:
`l` is actually called once on the line before, and then, if we're at the end of the line, it's called again so Atom takes care of going to the next line for us.